waspc/src/Wasp/Generator/ServerGenerator/EmailSenderG.hs

```diff
@@ -26,7 +28,12 @@ genEmailSender spec = case maybeEmailSender of
       <++> genCore emailSender
   Nothing -> return []
   where
-    maybeEmailSender = AS.App.emailSender $ snd $ getApp spec
+    maybeEmailSender = AS.App.emailSender (snd $ getApp spec) <|> maybeDummyEmailSender
+    maybeDummyEmailSender =
```

Comment:
**Only in development** we provide the default dummy email sender.

Reply:
Ok, so if they haven't specified any email sender, and we are in development, then we behave as if they specified the "dummy email sender".

But they can also specify it manually, right? They can write `emailSender: Dummy` in main.wasp?

When I think about us assuming they want to use dummy when they don't specify anything, I see a potential problem -> maybe they are doing some actual email sending, and even though they haven't specified any real email sender, they are using Wasp's dummy email sender, not maybe even knowing for real that they are using it -> stuff just works, and emails are written in in the console, ok cool. Then they go to deploy, and it will get deployed indeed, but it will crash in production when email is to be sent, because there is no email sender. SOmething like that? Or will it fail during `wasp build` because some code is missing? How will that go?

I wonder if we should require them to always be explicit and do `emailSender: Dummy`. That way we know they want to use email sender, and we can tell them "use the real one now please, because you want to go to production", and we can crash `wasp build` with such nice message.
Yes, this means that when using email auth they have to add this one extra line, but actually that might be good, to make them aware that this is happening.

Reply:
> Then they go to deploy, and it will get deployed indeed, but it will crash in production when email is to be sent, because there is no email sender. SOmething like that? Or will it fail during wasp build because some code is missing? How will that go?

We only provide the dummy email sender implicitly when it's **not a build** therefore, as soon as they try building for deployment, the Wasp compiler would crash with an error that they need to set up an email sender.

Reply:
Hmmmm I'm not sure about what is the better option, if the email sender is explicit but dummy, it works in dev and they are aware they need to change something when they want to get serious. That makes sense to me as well.

Reply:
We concluded in live communication to go with explicit!

Reply:
I've now updated the implementation to require setting the `Dummy` email sender explicitly by the user. 

If they attempt to deploy an app with the `Dummy` sender set, developers will get an error.